storybookjs/storybook
# Addon-docs: Fix exotic React components in Source block

## The Problem

React Memo & ForwardRef exotic components would show up in the code snippet of storybook as `<[object Object]>` because it didn't know where to lookup the inner render function's name.

This is a bug in the library Storybook uses to render JSX into a string: https://github.com/algolia/react-element-to-jsx-string/issues/352

## What I did

I stole the default `displayNameFn` from the original library [here](https://github.com/algolia/react-element-to-jsx-string/blob/4974512273c86c72415376fea89f9d6e07e2b2e5/src/parser/parseReactElement.js#L15) 
And added what it needed to fetch Memo & ForwardRef component's names.

## How to test

- Is this testable with Jest or Chromatic screenshots?
Maybe? I have no idea
- Does this need a new example in the kitchen sink apps?
No
- Does this need an update to the documentation?
No

If your answer is yes to any of these, please make sure to include it in your PR.
